lxqtbacklight/linux_backend/driver/libbacklight_backend.c

```diff
@@ -179,11 +191,10 @@ char *lxqt_backlight_backend_get_driver()
             if( !strcmp(dp->d_name, ".") || !strcmp(dp->d_name, "..") )
                 continue;
             driver = dp->d_name;
-            sprintf(path, "/sys/class/backlight/%s/type", driver);
-            FILE *in = fopen(path, "r");
+            FILE *in = open_driver_file("type", driver, "r");
             if( in == NULL )
                 continue;
-            int ok = fscanf(in, "%s", type);
+            int ok = fscanf(in, "%1024s", type);
```

Comment:
The documentation states that:
```
s      Matches  a  sequence  of  non-white-space  characters; the next pointer must be a pointer to
the initial element of character array that is long  enough to hold the input sequence and the
terminating null byte ('\0'), which is added automatically. 
```
so in order to accommodate the null byte shouldn't one write:
``` cpp
int ok = fscanf(in, "%1023s", type);
```
?

Reply:
True. The exact section from `man 3 fscanf` that explains it is this one:

```
An optional decimal integer which specifies the maximum field width. [...]. String
input conversions store a terminating null byte ('\0') to mark the end of the
input; the maximum field width does not include this terminator.
```
I will change it to `%1023s`

Reply:
I rebased the PR accordingly.